### PR TITLE
Search provider - NASA ADS

### PR DESCRIPTION
**Pull request by vilhelmp**
Rewrote the NASA ADS search I had lying around for inclusion into astroquery. Instead of parsing the HTML results with Beautiful Soup, I request the results in XML, and use the `xml.dom.minidom` inspector to parse the results.

Simple search:
```python    
    import astroquery.nasa_ads as ads
    result  = ads.ADS.query_simple('^Ginsburg')
    # result is now a Astropy.Table table. eg:
    result['authors'] # to see the list of lists of authors
```

However, `print result` raises an error, I think it's because of the encoding, in the function `astroquery.nasa_ads.utils.get_data_from_xml` the parsing of the individual elements happens (`.encode("utf-8")` tries to encode the text field into utf-8).

### CODE REVIEW

**Comment by keflavich** [comment]
For tables, you can use the function `astropy.table.Table.convert_bytestring_to_unicode` to fix some unicode issues, perhaps - I can't promise this is relevant to the problem you've noted, I'm just trying to pass on something I recently learned.  

Thanks for the PR.  I'll try to review it in more detail when I'm back from travel

**Comment by vilhelmp** [comment]
I just discovered this: https://github.com/andycasey/ads

**Comment by keflavich** [comment]
It is nice but we need an API key to use that.  Maybe we can get one for
astroquery?  I think a python interface to the simpler, old ADS form is
still useful

On Tue, Mar 17, 2015 at 11:18 AM, Magnus Vilhelm Persson <
notifications@github.com> wrote:

> I just discovered this: https://github.com/andycasey/ads
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/astropy/astroquery/pull/499#issuecomment-82376496>.
>



-- 
Adam

**Comment by keflavich** [comment]
Can you rebase or merge against master?  we fixed the astropy build issue in #503

**Comment by vilhelmp** [comment]
I think I did it now, never used the `rebase` option before.

**Comment by keflavich** [comment]
@vilhelmp There is one error related to the docs - add the new NASA_ADS documentation file to `index.rst`

**Comment by vilhelmp** [comment]
do I have to rebase and all that again to do this?

**Comment by vilhelmp** [comment]
Yeah I don't know what I just did. I tried to rebase again, before adding the doc to index.rst. Got an error, that there were conflicts, from the first commit. Then I thought, I clone my fork again locally (to a new destination), and then work on that instead, that seem to have screwed up things on my machine, now I don't know what to do.

**Comment by keflavich** [comment]
I'll investigate tomorrow.

**Comment by keflavich** [comment]
@vilhelmp if you want some live help dealing with the git-related stuff, join gitter (https://gitter.im/astropy/astropy) and I can help walk you through the issues.  Re-cloning should never be necessary, though I did it a few times before I understood rebasing.

As for the error, it's still a simple one:
https://travis-ci.org/astropy/astroquery/jobs/58165454#L1473
just need to add the NASA ADS entry in the index.

**Comment by vilhelmp** [comment]
I get some errors in other modules, probably because I updated my fork from upstream, and then pushed my changes.

**Comment by keflavich** [comment]
Travis is passing.  Coverage is complaining because you don't have tests, but that isn't a blocker - tests would be really great, but I'm willing to incorporate un-locally-tested tools for the time being at least.  So, is this ready to merge, from your side?

**Comment by keflavich** [comment]
Actually, on further inspection, we need some docs and at least a remote test/example...

File: docs/nasa_ads/nasa_ads.rst [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+This module provides an interface to the online `SAO/NASA Astrophysics Data System`_.
+At the moment only the "simple search", i.e. omni-box search is available, and only 
+a subset of the results are accessible.
+
+Examples
+========
+
+Search works by specific identifier
+-----------------------------------
+
```
Comment (keflavich) [inline docs/nasa_ads/nasa_ads.rst]: this stuff should be in an indented code block:

```
.. code-block:: python

    from astro....
```

**Comment by keflavich** [comment]
the failure is just sphinx

**Comment by keflavich** [comment]
you're green now!  ready to merge?

File: astroquery/nasa_ads/__init__.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        'journal', 'volume', 'pubdate', 'page', 'lastpage', 'keywords', 'keyword',
+        'origin', 'copyright', 'link', 'name', 'url', 'count', 'score', 'citations',
+        'abstract', 'doi', 'eprintid']
+
+from .core import ADSClass, ADS
+
+__all__ = ['ADSClass', 'ADS',
+        'Conf', 'conf']
+
+#~ class Conf(_config.ConfigNamespace):
```
Comment (keflavich) [inline astroquery/nasa_ads/__init__.py]: all these commented out things should be removed

File: astroquery/nasa_ads/core.py [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+
+:author: Magnus Persson <magnusp@vilhelm.nu>
+
+"""
+
+import warnings
+# example warning 
+# warnings.warn("Band was specified, so blabla is overridden")
+#~ from astropy.io import ascii
+#~ from astropy import units as u
```
Comment (keflavich) [inline astroquery/nasa_ads/core.py]: remove comments

File: astroquery/nasa_ads/core.py [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+        # i.e. a dictionary
+        return {'qsearch' : query_string, 'data_type' : 'XML'}
+
+
+
+ADS = ADSClass()
+
+
+"""
+typical fields available:
```
Comment (keflavich) [inline astroquery/nasa_ads/core.py]: could make this list an ADS property `available_fields` ?

File: astroquery/nasa_ads/utils.py [hunk 5]
```diff
@@ -0,0 +1,16 @@
+
+
+def get_data_from_xml(doclist, fieldname, nohitreturn=None):
```
Comment (keflavich) [inline astroquery/nasa_ads/utils.py]: docstring

Comment (keflavich) [inline astroquery/nasa_ads/utils.py]: and if this is a private utility, maybe make it `_get_data...` ?

**Comment by vilhelmp** [comment]
Think I fixed everything!

**Pull request merged**